### ISSUE TITLE
Add a random_peer() method to Peerstore

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#316]
Allows choosing a random peer from the store.

Alternatively, something like a combination of `num_peers()` and `nth_peer()` could work as well.

This is not urgent, so no need to rush a bad API.

**Comment by tomaka** [comment]
The method shouldn't return an "expired" node, ie. whose multiaddress have all expired.

**Comment by tomaka** [comment]
No longer relevant after #723
